lib/ansible/modules/packaging/os/bindep.py

```diff
@@ -0,0 +1,189 @@
+        - libffi-dev [platform:dpkg test]
+        - libffi-devel [platform:rpm test]
+
+# Get the list of packages to install on a given host for the test profile
+- bindep:
+    requirements:
+        - build-essential [platform:dpkg test]
+        - gcc [platform:rpm test]
+        - language-pack-en [platform:ubuntu]
+        - libffi-dev [platform:dpkg test]
+        - libffi-devel [platform:rpm test]
+    profiles:
+        - test
+'''
+
+
+RETURN = '''
+bindep_packages:
+    description: Dictionary containing information about the requirements.
+    returned: On success.
+    type: dictionary
+    contains:
+        missing:
+            description: Packages that are missing from the system
+            returned: success
+            type: list
+            sample:
+                - libmysqlclient-dev
+                - libxml2-dev
+        badversion:
+            description: Packages that are installed but at bad versions.
+            returned: success
+            type: list
+            sample:
+                - {'package': 'libxml2-dev', 'version': '2.9.4+dfsg1-2, 'constraint': '>= 3.0'}
+        up_to_date:
+            description: Flag indicating all packages are up to date
+            returned: success
+            type: bool
+'''
+
+import os
+
+from ansible.module_utils.basic import AnsibleModule
+
+try:
+    import bindep.depends
+    import ometa.runtime
+    HAS_BINDEP = True
+except ImportError:
+    HAS_BINDEP = False
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            path=dict(required=False, type='str'),
+            requirements=dict(required=False, type="list"),
+            profiles=dict(required=False, default=['default'], type='list'),
+        ),
+        mutually_exclusive = [['path','requirements']],
+        required_one_of = [['path','requirements']],
+        supports_check_mode = True
+    )
+
+    if not HAS_BINDEP:
+        module.fail_json(msg='bindep is required for this module')
+    if not hasattr(bindep.depends, 'get_depends'):
+        module.fail_json(
+            msg='bindep is required at version >= 2.2 for this module')
+
+    path = module.params['path']
+    requirements = module.params['requirements']
+    profiles = module.params['profiles']
+
+    try:
+        if requirements:
+            req_string = '\n'.join(requirements) + '\n'
+            depends = bindep.depends.Depends(req_string)
+        else:
+            if path and os.path.isdir(path):
+                os.chdir(path)
+                depends = bindep.depends.get_depends()
+                if not depends:
+                    module.fail_json(
+                        msg="no bindep.txt file found at %s" % path)
+            else:
+                depends = bindep.depends.get_depends(filename=path)
+                if not depends:
+                    module.fail_json(msg="bindep file %s not found" % path)
+    except ometa.runtime.ParseError as e:
```

Comment:
That is due to py2.4 syntax check failing.  Assuming this module won't support py24, the path will need added to `test/compile/python2.4-skip.txt`

Otherwise, check http://docs.ansible.com/ansible/dev_guide/developing_modules_python3.html#exceptions for how to handle exceptions that support py2.4-py3